graph-builder/src/registry.rs

```diff
@@ -121,17 +124,35 @@ pub fn fetch_releases(
     let tagged_layers = thread_runtime.block_on(tags)?;
 
     let mut releases = Vec::with_capacity(tagged_layers.len());
-    for (tag, layer_digests) in tagged_layers {
-        let release = cache_release(
+    for (tag, manifestref, layer_digests) in tagged_layers {
+        if let Some(release) = cache_release(
             layer_digests,
             authenticated_client.to_owned(),
             registry_host.to_owned(),
             repo.to_owned(),
             tag.to_owned(),
             cache,
-        )?;
-        if let Some(metadata) = release {
-            releases.push(metadata);
+        )? {
+            let manifestref = manifestref.as_str();
+            let quay_labels = thread_runtime.block_on(quay_client.fetch_labels(
+                &repo,
+                &manifestref,
+                Some(&quay_label_filter),
+            ))?;
+
+            let release = Release {
+                metadata: Metadata {
+                    metadata: quay_labels
+                        .into_iter()
+                        .map(|label| (label.key, label.value))
+                        .chain(release.metadata.metadata)
```

Comment:
I think these two iterators should be chained in the reverse order. I think with the current logic, if the same label key exists in both image manifest and quay API, the image one is inserted last and kept.

(This is from a quick reading of this code and my current understanding of hashmap collection. Please double-check my assertion as either one of those could be wrong.)

Reply:
Nice catch. I meant to revisit this before merging but who knows if I would've remembered today since I forgot to add a TODO for it too ;-) If we want to the labels to have precedence -- which could be useful even though I can't come up with a immediate use case now -- your suggestion is correct and we need to reverse them. See [this playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=402ad76ed37ad002397bdaf00ced72b4).

Reply:
I think it makes sense to state that mutable labels have precedence over image ones. The usecase I had in mind was "amend a mistake in an (immutable) image metadata label, via a (mutable) quay label with the same key-name".